deployment/auto_scaler.go

```diff
@@ -0,0 +1,105 @@
+package deployment
+
+import (
+	"fmt"
+	"time"
+	"v9_deployment_manager/database"
+	"v9_deployment_manager/log"
+	"v9_deployment_manager/worker"
+)
+
+type AutoScaler struct {
+	actionManager *ActionManager
+	driver        *database.Driver
+	workers       []*worker.V9Worker
+}
+
+type ComponentStatAndInstances struct {
+	instanceCount int
+	averageStats  worker.ComponentStats
+}
+
+var MaxHits = 100.0
+var MinHits = 15.0
+
+func (scaler *AutoScaler) AutoScale() {
+	//Get formatted worker ids
+	// TODO: Pull this out into a helper function
+	workerIDs := make([]string, len(scaler.workers))
+	for i := range scaler.workers {
+		name := fmt.Sprintf("worker_%d", i)
+		id, err := scaler.driver.FindWorkerID(name)
+		if err != nil {
+			log.Error.Println("error getting worker id:", err)
+			continue
+		}
+		workerIDs[i] = id
+	}
```

Comment:
We now calculate this "worker name" info in a bunch of places. We should pull it out to a helper for sure